### PR TITLE
Revamped requirements

### PR DESCRIPTION
**Pull request by mlabuda**

### CODE REVIEW

**Comment by genie-reddeer** [comment]
Can one of the admins verify this patch?

**Comment by rawagner** [comment]
testPR

**Comment by mlabuda** [comment]
Wiki page to new requirements https://github.com/eclipse/reddeer/wiki/Requirements-2.0
plus we have to resolve following:
- what to log and format of logging
- nested configurations

**Comment by mlabuda** [comment]
Ok, added support for nested configurations. Amended and force-pushed commit with changes. Created follow-ups #1745 and #1746 which should be addressed ASAP (at least as soon as this PR is merged; before RD 2.0). Job passed green(before amending), yey! 

@rawagner  @apodhrad @odockal please review. PR is ready to review.

**Comment by mlabuda** [comment]
testPR

**Comment by odockal** [comment]
testPR

File: plugins/org.eclipse.reddeer.junit/src/org/eclipse/reddeer/junit/requirement/matcher/RequirementMatcher.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import org.hamcrest.BaseMatcher;
+
+/**
+ * General Requirement Attribute matcher. Useful for matching requirement's 
+ * attribute to a matching value.
+ * 
+ * @author mlabuda@redhat.com
+ *
+ */
+public abstract class RequirementMatcher extends BaseMatcher<String> {
```
Comment (rawagner) [inline plugins/org.eclipse.reddeer.junit/src/org/eclipse/reddeer/junit/requirement/matcher/RequirementMatcher.java]: can we rename this class to AbstractRequirementMatcher ?

**Comment by rawagner** [comment]
I have created a simple test with @ApacheTomcatServer requirement and added @RequirementRestriction for version >9.0 --> which should not match any of defined servers in json config --> which is OK. However a test should fail (requirement cannot be fulfilled) but it is completely ignored.

**Comment by rawagner** [comment]
pushed to master

**Pull request closed without merging**